import sys
from io import StringIO
from contextlib import redirect_stdout, redirect_stderr
import json
import traceback

{{ user_code | safe }}

def run_solution():
    results = []
    for test_case in {{ test_cases | tojson | safe }}:
        error = ""
        logged_output = StringIO()
        logged_error = StringIO()

        input_data = test_case['input']
        expected_output = test_case['output']
        return_output = None
        valid = None
        
        try:
            with redirect_stdout(logged_output), redirect_stderr(logged_error):
                {{ problem_test_run_code | safe | indent(16) }}

        except Exception as e:
            error = traceback.format_exc()
        
        results.append({
            "input": input_data,
            "expected": expected_output,
            "output": return_output,
            "valid": valid,
            "error": error,
            "std_output": logged_output.getvalue().strip(),
            "std_error": logged_error.getvalue().strip()
        })
    return {"results": results}

if __name__ == "__main__":
    print(json.dumps(run_solution()))